modules/wallet/wallet.go

```diff
@@ -68,8 +68,9 @@ type Wallet struct {
 	// from the seeds, when checking new outputs or spending outputs, the seeds
 	// are not referenced at all. The seeds are only stored so that the user
 	// may access them.
-	seeds []modules.Seed
-	keys  map[types.UnlockHash]spendableKey
+	seeds      []modules.Seed
+	keys       map[types.UnlockHash]spendableKey
+	futureKeys map[types.UnlockHash]bool
```

Comment:
I'm thinking this should map to the seed index, not a bool, so that you can easily jump forward to the correct index when you see a future key.